Initial windows support for certificate store (#1575)
grafana/agent: docs/user/configuration/server-config.md

# File path to the signing CA certificate, needed if CA is not trusted
[client_ca_file: <string>]

# Windows certificate filter allows selecting client CA and server certificate from the Windows Certificate store
[windows_certificate_filter: <windows_certificate_fillter_config>]
```

## windows_certificate_filter

The `windows_certificate_filter` configures usage of the Windows Certificate store
NOTE: cert_file, key_file, and client_ca_file are not used when this is enabled

```yaml

# Name of the system store to look for the Client Certificate ex LocalMachine, CurrentUser 

> [client_system_store: <string>]

Author: I don't think we can define a default value, it's likely LocalMachine, but it could also be the Services location too.

# Name of the store to look for the Client Certificate ex My, CA
[client_store: <string>]

# Array of issuer common names to check against
[client_issuer_common_names: [<string>]]

# Regular expression to match Subject name
[client_subject_regex: <string>]

# Client Template ID to match in ASN1 format ex "1.2.3"

> [client_template_id: <string>]

Author: Took a look and it is cleaner to go separate blocks so done!

# File path to the signing CA certificate, needed if CA is not trusted
[client_ca_file: <string>]

# Windows certificate filter allows selecting client CA and server certificate from the Windows Certificate store
[windows_certificate_filter: <windows_certificate_filter_config>]
```

## windows_certificate_filter_config

The `windows_certificate_filter_config` configures the use of the Windows Certificate store. Setting cert_file, key_file, and client_ca_file are invalid settings when using the windows_certificate_filter.

```yaml
# Client configuration, optional. If nothing specific will use the default client ca root
[client: <windows_client_config>]
  
# Name of the store to look for the Client Certificate ex My, CA
[server: <windows_server_config>]
```

### windows_client_config

```yaml
# Name of the system store to look for the Client Certificate ex LocalMachine, CurrentUser 
[system_store: <string>]

# Name of the store to look for the Client Certificate ex My, CA

> [store: <string>]

Reviewer: Are we sure these aren't required? What will happen if there's no system store / store set?

## windows_certificate_filter

> The `windows_certificate_filter` configures usage of the Windows Certificate store. It is invalid to set cert_file, key_file, and client_ca_file when using windows_certificate_filter.

Reviewer 2: ```suggestion
The `windows_certificate_filter` configures the use of the Windows Certificate store. Set cert_file, key_file, and client_ca_file are invalid settings when using the windows_certificate_filter.
```
